bench/many_flat.html

```diff
@@ -0,0 +1,44 @@
+<!DOCTYPE html>
+<html>
+
+<head></head>
+
+<body>
+  <script type="module">
+    import { hydrateShadowRoots as manualWalkHydration } from '../_implementation/manual_walk.js';
+    import { hydrateShadowRoots as querySelectorHydration } from '../_implementation/queryselectorall.js';
+
+    function benchmark(hydrateMethod) {
+      let hydrateShadowRoots;
+      if (hydrateMethod === 'manualWalk') {
+        hydrateShadowRoots = manualWalkHydration;
+      } else if (hydrateMethod === 'querySelector') {
+        hydrateShadowRoots = querySelectorHydration;
+      } else {
+        return 9 * 1000 * 1000;
+      }
+      const html = `
```

Comment:
Another nit, but along with the recursion comment above, I think a single template could generation both of these:

Something like....?

```js
const generate = (length, depth, i = 0, d = 0) => {
  if (i >= length || d >= depth) return '';
  return html`
    <div>
      <template shadowroot="open">
        ${generate(length, depth, 0, d+1)}
        <div>Hello world</div>
      </template>
    </div>
    ${generate(length, depth, i+1, d)}
  `;
}
```

Reply:
Don't wanna generalize over something so inchoate